--- OpenShadingLanguage-1.15.5.0.orig/src/cmake/externalpackages.cmake	2026-06-01 19:33:29.000000000 +0200
+++ OpenShadingLanguage-1.15.5.0/src/cmake/externalpackages.cmake	2026-07-26 17:39:21.411482217 +0200
@@ -58,7 +58,7 @@ checked_find_package (pugixml REQUIRED
 # LLVM library setup
 checked_find_package (LLVM REQUIRED
                       VERSION_MIN 14.0
-                      VERSION_MAX 22.9
+                      VERSION_MAX 23.9
                       PRINT LLVM_SYSTEM_LIBRARIES CLANG_LIBRARIES
                             LLVM_SHARED_MODE)
 # ensure include directory is added (in case of non-standard locations
--- OpenShadingLanguage-1.15.5.0.orig/src/liboslexec/llvm_util.cpp	2026-06-01 19:33:29.000000000 +0200
+++ OpenShadingLanguage-1.15.5.0/src/liboslexec/llvm_util.cpp	2026-07-26 17:39:21.447349719 +0200
@@ -78,7 +78,10 @@ OSL_GCC_PRAGMA(GCC diagnostic ignored "-
 #include <llvm/Transforms/Scalar.h>
 #include <llvm/Transforms/Scalar/GVN.h>
 #include <llvm/Transforms/Utils.h>
+#if OSL_LLVM_VERSION < 230
+// Removed in LLVM 23
 #include <llvm/Transforms/Utils/UnifyFunctionExitNodes.h>
+#endif
 
 #include <llvm/Support/DynamicLibrary.h>
 
